core/trino-main/src/main/java/io/trino/spiller/FileSingleStreamSpiller.java

```diff
@@ -168,7 +167,7 @@ private void writePages(Iterator<Page> pageIterator)
 
         try {
             InputStream input = closer.register(targetFile.newInputStream());
-            Iterator<Page> pages = PagesSerdeUtil.readPages(serde, new InputStreamSliceInput(input, BUFFER_SIZE));
```

Comment:
here `reads` still go to OS.

Reply:
The number of reads shouldn't increase as long as the average page size is above `8kb`. Is this generally the case for spilling?